src/test/spec/unified_runner/operation.rs

```diff
@@ -343,39 +309,65 @@ impl TestOperation for DeleteOne {
 }
 
 #[derive(Debug, Default, Deserialize)]
-#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
 pub(super) struct Find {
-    filter: Option<Document>,
+    filter: Document,
     session: Option<String>,
     #[serde(flatten)]
     options: FindOptions,
 }
 
+impl Find {
+    async fn get_cursor<'a>(
+        &'a self,
+        id: &'a str,
+        test_runner: &'a mut TestRunner,
+    ) -> Result<FindCursor> {
+        let collection = test_runner.get_collection(id).clone();
+        match &self.session {
+            Some(session_id) => {
+                let session = test_runner.get_mut_session(session_id);
+                let cursor = collection
+                    .find_with_session(Some(self.filter.clone()), self.options.clone(), session)
+                    .await?;
+                Ok(FindCursor::Session {
+                    cursor,
+                    session_id: session_id.clone(),
+                })
+            }
+            None => {
+                let cursor = collection
+                    .find(Some(self.filter.clone()), self.options.clone())
+                    .await?;
+                Ok(FindCursor::Normal(Mutex::new(cursor)))
+            }
+        }
+    }
+}
+
 impl TestOperation for Find {
     fn execute_entity_operation<'a>(
         &'a self,
         id: &'a str,
         test_runner: &'a mut TestRunner,
     ) -> BoxFuture<'a, Result<Option<Entity>>> {
         async move {
-            let collection = test_runner.get_collection(id).clone();
-            let result = match &self.session {
-                Some(session_id) => {
-                    let session = test_runner.get_mut_session(session_id);
-                    let mut cursor = collection
-                        .find_with_session(self.filter.clone(), self.options.clone(), session)
-                        .await?;
+            let result = match self.get_cursor(id, test_runner).await? {
+                FindCursor::Session {
+                    mut cursor,
+                    session_id,
+                } => {
+                    let session = test_runner.get_mut_session(&session_id);
                     cursor
                         .stream(session)
                         .try_collect::<Vec<Document>>()
                         .await?
                 }
-                None => {
-                    let cursor = collection
-                        .find(self.filter.clone(), self.options.clone())
-                        .await?;
+                FindCursor::Normal(cursor) => {
+                    let cursor = cursor.into_inner();
                     cursor.try_collect::<Vec<Document>>().await?
                 }
+                FindCursor::Closed => panic!("get_cursor returned a closed cursor"),
```

Comment:
`Find::get_cursor` *could* return a different enum that doesn't have the `Closed` variant and impls `Into<FindCursor>`, but that didn't seem worth the type proliferation for a single use.